pkg/controller/nodeipam/ipam/controller.go

```diff
@@ -92,6 +92,9 @@ func NewController(
 		return nil, err
 	}
 
+	if _, err := c.set.AllocateNext(); err != nil {
```

Comment:
don't you need to dealloc

Reply:
Also, there should be log with error and I would return a new error type.

Reply:
Done. There is no error log in the NewController function. Can I return a new error type without printing log?